#Origionally from https://gitlab.com/gitlab-org/charts/gitlab-runner/-/merge_requests/191/diffs.patch
#Added extraVolumeMounts for init container too.

From 2d1c08c08f48fd31ddda05ce3dfe15cf465a4a56 Mon Sep 17 00:00:00 2001
From: Valdis Rigdon <valdis.rigdon@appian.com>
Date: Mon, 22 Jul 2019 11:22:54 -0400
Subject: [PATCH] Adds templating for extra containers, volumes, and
 volumeMounts

This gives the chart the ability to run extra containers as part of the
Deployment which can generate files to be shared into the gitlab-runner
container.
---
 templates/deployment.yaml | 9 +++++++++
 values.yaml               | 9 +++++++++
 2 files changed, 18 insertions(+)

diff --git a/templates/deployment.yaml b/templates/deployment.yaml
index b3512a99..04dedf81 100644
--- a/templates/deployment.yaml
+++ b/templates/deployment.yaml
@@ -61,6 +61,9 @@
           readOnly: true
           mountPath: /home/gitlab-runner/.gitlab-runner/certs/
         {{- end }}
+        {{- if .Values.extraVolumeMounts }}
+{{ toYaml .Values.extraVolumeMounts | indent 8 }}
+        {{- end }}
         resources:
 {{ toYaml .Values.resources | indent 10 }}
       serviceAccountName: {{ if .Values.rbac.create }}{{ include "gitlab-runner.fullname" . }}{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}
@@ -108,6 +108,12 @@ spec:
           readOnly: true
           mountPath: /home/gitlab-runner/.gitlab-runner/certs/
         {{- end }}
+        {{- if .Values.extraVolumeMounts }}
+{{ toYaml .Values.extraVolumeMounts | indent 8 }}
+        {{- end }}
+      {{- if .Values.extraContainers }}
+{{ toYaml .Values.extraContainers | indent 6 }}
+        {{- end }}
         resources:
 {{ toYaml .Values.resources | indent 10 }}
       volumes:
@@ -147,6 +153,9 @@ spec:
       - name: scripts
         configMap:
           name: {{ include "gitlab-runner.fullname" . }}
+      {{- if .Values.extraVolumes }}
+{{ toYaml .Values.extraVolumes | indent 6 }}
+      {{- end }}
     {{- if .Values.affinity }}
       affinity:
 {{ toYaml .Values.affinity | indent 8 }}
diff --git a/values.yaml b/values.yaml
index 7ece3544..efc2335f 100644
--- a/values.yaml
+++ b/values.yaml
@@ -363,3 +363,12 @@ podLabels: {}
   #   pods:
   #     metricName: gitlab_runner_jobs
   #     targetAverageValue: 400m
+
+## extra containers to run
+extraContainers: {}
+
+## extra volume mounts for the gitlab-runner container
+extraVolumeMounts: {}
+
+## extra volumes
+extraVolumes: {}
-- 
2.24.1

